core/src/main/resources/struts-default.xml

```diff
@@ -377,6 +378,10 @@
                 <interceptor-ref name="alias"/>
                 <interceptor-ref name="servletConfig"/>
                 <interceptor-ref name="i18n"/>
+                <interceptor-ref name="cspInterceptor">
+                    <param name="enforcingMode">false</param>
+                    <param name="reportUri">/csp-reports</param>
```

Comment:
What will happen if `reportUri` is set and no action is configured? Maybe we should remove `reportUri` from default xml?

Reply:
A user can still configure the `reportUri` value to point to an external host! So even if devs choose not to use/subclass the action that Struts provides, they can still use CSP's reporting features (see services like https://report-uri.io and suchlike). I would say leaving it here gives it visibility and perhaps we can make sure that documentation is clear on this point? 

WDYT?

Reply:
This is the default stack, meaning that report uri will be set for all applications using it. If there is no `/csp-reports` action then what will happen? 404?

Also (slim chances but still) what if application already has action with that exact name, but for some other stuff?

Reply:
That's a fair point! I've pushed a new commit with these changes so the new behaviour is:

- By default, CSP will be set to report-mode (so all websites continue working as previously) and violations will only show up on the console.
- If a user decides to add a reportUri, violations will be seen on the console and will be sent to the URI.
- A user may choose to turn enforcement mode on and the above two points still hold.

I think this is a less intrusive default config!